File: backend/foodapp/foodapp/Models/MenuItem.cs
namespace foodapp.Models
{
    public class SubItem
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public double Price { get; set; }
    }

    public class MenuItem
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string ImageUrl { get; set; } = string.Empty;
        public List<SubItem> SubItems { get; set; } = new();
    }
}

File: lib/models/menu_item.dart
class SubItem {
  final int? id;
  final String name;
  final double price;

  SubItem({this.id, required this.name, required this.price});

  factory SubItem.fromJson(Map<String, dynamic> json) {
    return SubItem(
      id: json['id'],
      name: json['name'],
      price: (json['price'] as num).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {if (id != null) 'id': id, 'name': name, 'price': price};
  }
}

class MenuItem {
  final int? id;
  final String title;
  final String imageUrl;
  final List<SubItem> subItems;

  MenuItem({
    this.id,
    required this.title,
    required this.imageUrl,
    this.subItems = const [],
  });

  factory MenuItem.fromJson(Map<String, dynamic> json) {
    return MenuItem(
      id: json['id'],
      title: json['title'],
      imageUrl: json['imageUrl'],
      subItems:
          (json['subItems'] as List<dynamic>)
              .map((e) => SubItem.fromJson(e as Map<String, dynamic>))
              .toList(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'title': title,
      'imageUrl': imageUrl,
      'subItems': subItems.map((e) => e.toJson()).toList(),
    };
  }
}
